x/chainlink/ante/ante.go

```diff
@@ -126,11 +126,30 @@ func (fd FeedDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, ne
 			if !feed.Feed.Empty() {
 				return ctx, sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, "feed already exists")
 			}
+		case *types.MsgAddFeedProvider:
+			feed := fd.chainLinkKeeper.GetFeed(ctx, t.GetFeedId())
+			if feed.Feed.Empty() {
+				return ctx, sdkerrors.Wrap(sdkerrors.ErrKeyNotFound, "feed does not exist")
+			}
+			if (types.DataProviders)(feed.GetFeed().GetDataProviders()).Contains(t.GetDataProvider().GetAddress()) {
+				return ctx, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "data provider already registered")
+			}
+		case *types.MsgRemoveFeedProvider:
+			feed := fd.chainLinkKeeper.GetFeed(ctx, t.GetFeedId())
+			if feed.Feed.Empty() {
+				return ctx, sdkerrors.Wrap(sdkerrors.ErrKeyNotFound, "feed does not exist")
+			}
+			if !(types.DataProviders)(feed.GetFeed().GetDataProviders()).Contains(t.GetAddress()) {
+				return ctx, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "data provider not present")
+			}
 		default:
 			continue
 		}
 	}
 
+	// TODO check feed owner #12
```

Comment:
ok, so you mean check if signer is the owner of the feed in #12?

Reply:
Yes ;)